File: plugins/agent-sdk-dev/examples/e2b-python/deploy/e2b_ephemeral.py
import os
import sys
import asyncio
from dotenv import load_dotenv
from e2b_code_interpreter import Sandbox
from pathlib import Path

# Add src to path
sys.path.insert(0, str(Path(__file__).parent.parent / "src"))
from agent import create_agent, run_agent


async def deploy_ephemeral(
    e2b_api_key: str,
    anthropic_api_key: str,
    task: str,
    template: str = "anthropic-claude-code",
    timeout: int = 300000
) -> str:
    """
    Deploy agent in ephemeral E2B sandbox.

    Args:
        e2b_api_key: E2B API key
        anthropic_api_key: Anthropic API key
        task: Task for the agent to perform
        template: E2B template to use
        timeout: Sandbox timeout in milliseconds

    Returns:
        Agent's response
    """

    print("Creating E2B sandbox (ephemeral mode)...")

    sandbox = Sandbox(template=template)

    try:
        print(f"Sandbox created: {sandbox.id}")
        print("Initializing agent in sandbox...")

        # Create agent
        agent = create_agent(api_key=anthropic_api_key)

        print("Running task...")
        result = await run_agent(agent, task)

        print("\n=== Agent Response ===")
        print(result)
        print("===================\n")

        return result

    except Exception as error:
        print(f"Deployment error: {error}")
        raise
    finally:
        print("Cleaning up sandbox...")
        sandbox.close()
        print("Sandbox closed")


async def main():
    """CLI handler for ephemeral deployment."""

    load_dotenv()

    e2b_api_key = os.getenv("E2B_API_KEY")
    anthropic_api_key = os.getenv("ANTHROPIC_API_KEY")

    if not e2b_api_key or not anthropic_api_key:
        print("Error: E2B_API_KEY and ANTHROPIC_API_KEY must be set")
        sys.exit(1)

    task = sys.argv[1] if len(sys.argv) > 1 else "Write a simple Python hello world script"
    template = os.getenv("E2B_TEMPLATE", "anthropic-claude-code")
    timeout = int(os.getenv("E2B_TIMEOUT", "300000"))

    try:
        await deploy_ephemeral(
            e2b_api_key=e2b_api_key,
            anthropic_api_key=anthropic_api_key,
            task=task,
            template=template,
            timeout=timeout
        )
        print("Deployment completed successfully")
    except Exception as error:
        print(f"Deployment failed: {error}")
        sys.exit(1)


if __name__ == "__main__":
    asyncio.run(main())

File: plugins/agent-sdk-dev/examples/e2b-typescript/deploy/e2b-ephemeral.ts
import 'dotenv/config'
import { Sandbox } from '@e2b/code-interpreter'
import { createAgent, runAgent } from '../src/agent.js'

interface DeploymentConfig {
  e2bApiKey: string
  anthropicApiKey: string
  template?: string
  timeout?: number
}

export async function deployEphemeral(
  config: DeploymentConfig,
  task: string
): Promise<string> {
  const {
    e2bApiKey,
    anthropicApiKey,
    template = 'anthropic-claude-code',
    timeout = 300000
  } = config

  console.log('Creating E2B sandbox (ephemeral mode)...')

  const sandbox = await Sandbox.create(template, {
    apiKey: e2bApiKey,
    timeout
  })

  try {
    console.log(`Sandbox created: ${sandbox.id}`)
    console.log('Initializing agent in sandbox...')

    // Create agent
    const agent = createAgent({ apiKey: anthropicApiKey })

    console.log('Running task...')
    const result = await runAgent(agent, task)

    console.log('\n=== Agent Response ===')
    console.log(result)
    console.log('===================\n')

    return result

  } catch (error) {
    console.error('Deployment error:', error)
    throw error
  } finally {
    console.log('Cleaning up sandbox...')
    await sandbox.close()
    console.log('Sandbox closed')
  }
}

// CLI handler
if (import.meta.url === `file://${process.argv[1]}`) {
  const e2bApiKey = process.env.E2B_API_KEY
  const anthropicApiKey = process.env.ANTHROPIC_API_KEY

  if (!e2bApiKey || !anthropicApiKey) {
    console.error('Error: E2B_API_KEY and ANTHROPIC_API_KEY must be set')
    process.exit(1)
  }

  const task = process.argv[2] || 'Write a simple Python hello world script'

  deployEphemeral(
    {
      e2bApiKey,
      anthropicApiKey,
      template: process.env.E2B_TEMPLATE,
      timeout: process.env.E2B_TIMEOUT ? parseInt(process.env.E2B_TIMEOUT) : undefined
    },
    task
  )
    .then(() => {
      console.log('Deployment completed successfully')
      process.exit(0)
    })
    .catch((error) => {
      console.error('Deployment failed:', error)
      process.exit(1)
    })
}
